tfhe/src/shortint/server_key/mod.rs

```diff
@@ -1685,3 +1496,65 @@ impl ServerKey {
             .min_by_key(|op| op.number_of_pbs())
     }
 }
+
+pub(crate) fn apply_blind_rotate(
+    bootstrapping_key: &ShortintBootstrappingKey,
+    in_buffer: &LweCiphertextView<u64>,
+    acc: &mut GlweCiphertextMutView<u64>,
+    buffers: &mut ComputationBuffers,
+) {
+    match &bootstrapping_key {
+        ShortintBootstrappingKey::Classic(fourier_bsk) => {
+            let fft = Fft::new(fourier_bsk.polynomial_size());
+            let fft = fft.as_view();
+            buffers.resize(
+                programmable_bootstrap_lwe_ciphertext_mem_optimized_requirement::<u64>(
+                    fourier_bsk.glwe_size(),
+                    fourier_bsk.polynomial_size(),
+                    fft,
+                )
+                .unwrap()
+                .unaligned_bytes_required(),
+            );
+            let stack = buffers.stack();
+
+            // Compute the blind rotation
+            blind_rotate_assign_mem_optimized(in_buffer, acc, fourier_bsk, fft, stack);
+        }
+        ShortintBootstrappingKey::MultiBit {
+            fourier_bsk,
+            thread_count,
+            deterministic_execution,
+        } => {
+            if *deterministic_execution {
+                multi_bit_deterministic_blind_rotate_assign(
+                    in_buffer,
+                    acc,
+                    fourier_bsk,
+                    *thread_count,
+                );
+            } else {
+                multi_bit_blind_rotate_assign(in_buffer, acc, fourier_bsk, *thread_count);
+            }
+        }
+    };
+}
+
+pub(crate) fn apply_bootstrap(
+    bootstrapping_key: &ShortintBootstrappingKey,
+    in_buffer: &LweCiphertextView<u64>,
+    out_buffer: &mut LweCiphertextMutView<u64>,
```

Comment:
maybe make generic for the buffer containers ?

Reply:
Done